buildkit/platform.go

```diff
@@ -32,7 +33,27 @@ func DetermineBuildPlatformFromHost() BuildPlatform {
 	return PlatformLinuxAMD64
 }
 
+func ParsePlatform(platformStr string) (BuildPlatform, error) {
+	if platformStr == "" {
+		return DetermineBuildPlatformFromHost(), nil
+	}
+
+	platform, err := platforms.Parse(platformStr)
+	if err != nil {
+		return BuildPlatform{}, fmt.Errorf("invalid platform format: %s. Must be one of: linux/amd64, linux/arm64, etc", platformStr)
+	}
+
+	return BuildPlatform{
```

Comment:
We could probably just remove `BuildPlatform` completely and use the interface defined by `platforms`, but this fixes the immediate papercut without reworking the code more.